python/kserve/kserve/model_server.py

```diff
@@ -13,95 +13,141 @@
+        model_repository_extension = ModelRepositoryExtension(model_registry=self.registered_models)
+        v1_endpoints = V1Endpoints(dataplane, model_repository_extension)
+        v2_endpoints = V2Endpoints(dataplane, model_repository_extension)
+
+        return FastAPI(
+            title="KServe ModelServer",
+            version=pkg_resources.get_distribution("kserve").version,
+            docs_url="/docs" if self.enable_docs_url else None,
+            redoc_url=None,
+            routes=[
+                # Server Liveness API returns 200 if server is alive.
+                FastAPIRoute(r"/", dataplane.live),
+                # Metrics
+                FastAPIRoute(r"/metrics", metrics_handler, methods=["GET"]),
+                # V1 Inference Protocol
+                FastAPIRoute(r"/v1/models", v1_endpoints.models, tags=["V1"]),
+                # Model Health API returns 200 if model is ready to serve.
+                FastAPIRoute(r"/v1/models/{model_name}", v1_endpoints.model_ready, tags=["V1"]),
+                FastAPIRoute(r"/v1/models/{model_name}:predict",
+                             v1_endpoints.predict, methods=["POST"], tags=["V1"]),
+                FastAPIRoute(r"/v1/models/{model_name}:explain",
+                             v1_endpoints.explain, methods=["POST"], tags=["V1"]),
+                # V2 Inference Protocol
+                # https://github.com/kserve/kserve/tree/master/docs/predict-api/v2
+                FastAPIRoute(r"/v2", v2_endpoints.metadata, tags=["V2"]),
+                FastAPIRoute(r"/v2/health/live", v2_endpoints.live, tags=["V2"]),
+                FastAPIRoute(r"/v2/health/ready", v2_endpoints.ready, tags=["V2"]),
+                FastAPIRoute(r"/v2/models/{model_name}", v2_endpoints.model_metadata, tags=["V2"]),
+                FastAPIRoute(r"/v2/models/{model_name}/versions/{model_version}",
+                             v2_endpoints.model_metadata, tags=["V2"], include_in_schema=False),
+                FastAPIRoute(r"/v2/models/{model_name}/infer",
+                             v2_endpoints.infer, methods=["POST"], response_model=InferenceResponse, tags=["V2"]),
+                FastAPIRoute(r"/v2/models/{model_name}/versions/{model_version}/infer",
+                             v2_endpoints.infer, methods=["POST"], tags=["V2"], include_in_schema=False),
+                FastAPIRoute(r"/v2/repository/models/{model_name}/load",
+                             v2_endpoints.load, methods=["POST"], tags=["V2"]),
+                FastAPIRoute(r"/v2/repository/models/{model_name}/unload",
+                             v2_endpoints.unload, methods=["POST"], tags=["V2"]),
+            ], exception_handlers={
+                errors.InvalidInput: errors.invalid_input_handler,
+                errors.InferenceError: errors.inference_error_handler,
+                errors.ModelNotFound: errors.model_not_found_handler,
+                errors.ModelNotReady: errors.model_not_ready_handler,
```

Comment:
wanna add another entry for `NotImplementedError`
mainly for 
-  Model versioning not supported yet
-  Explanation endpoint not supported yet